DOCS-386 - Add Cloud SOAR courses to doc (#4747)
SumoLogic/sumologic-documentation: docs/cloud-soar/introduction.md

---
id: introduction
title: Introduction to Cloud SOAR
sidebar_label: Introduction
---

import useBaseUrl from '@docusaurus/useBaseUrl';

> This article provides an introduction to Cloud SOAR for analysts and administrators.

Reviewer: A suggestion -- if the user isn't sure of their role (analyst vs. admin), they could find out [here](https://help.sumologic.com/docs/get-started/onboarding-checklists/).

Author: Great suggestion, @kimsauce!

Custom playbooks allow you to automate any task that uses a custom API. You can also use them to automate tasks that aren’t part of the hundreds of default playbooks included in Cloud SOAR.

Playbooks are made up of nodes which are connected together in a flowchart. Whether you’re customizing a playbook or creating one from scratch, you have several node options:
* **Actions**. Enrich data, execute APIs, send notifications, or use other integrations.
* **Tasks**. Assign a task to an analyst or another human for later review.
* **Conditions**. If-then statements that allow playbooks to branch in different directions.
* **User choice**. Pause the playbook and wait for a human’s decision.
* **Embedded playbooks**. Run another playbook.

Each of these nodes are color-coded in a playbook:
* A. **Conditions**. Conditions are represented by a purple diamond, allow your playbook to branch in different directions based on an if-then statement.
* B. **Enrichments**. Green nodes are enrichments. These might add additional information from a threat intel database or convert data from one type to another. 

> * C. **User Choice** User choices, represented by a red circle, pause a playbook and wait for a human's decision. After the choice is made,  the playbook can continue branching in different directions.

Reviewer: ```suggestion
* C. **User Choice**. User choices, represented by a red circle, pause a playbook and wait for a human's decision. After the choice is made,  the playbook can continue branching in different directions.
```

##### Best practices

Before you begin creating or customizing a playbook, decide what you’d like to automate. Think about what conditions you want met, and what actions or integrations you want to accomplish based on different flows. Once you have a design in mind for the flow of your playbook, you can create or customize a new one. Search App Central to see if an out-of-the-box playbook that does what you want already exists, or if you can modify a existing playbook that’s similar to what you have in mind. 

#### Create a custom playbook for Cloud SIEM Insights

Cloud SOAR allows us to create automations that will run whenever Cloud SIEM Insights are created or closed.  These automations are powered through playbooks, predefined actions run in an automated workflow to respond to an incident.  

Let’s create a playbook for use in Cloud SIEM.

1. [**Classic UI**](/docs/get-started/sumo-logic-ui-classic).  In the main Sumo Logic menu, select **Automation > Playbooks**. <br/>[**New UI**](/docs/get-started/sumo-logic-ui). In the main Sumo Logic menu, select **Automation > Playbooks**. You can also click the **Go To...** menu at the top of the screen and select **Playbooks**. 

> 1. You can click on any of the existing playbooks which will open the playbook diagram in the sidebar on the right. You can view here the individual nodes and sequences in the selected playbooks, to give you an idea of the type of actions and structures that you can create with custom playbooks. Playbooks can have any number of actions, as well as branching conditions to manage different sequences of actions, depending on selected criteria.  You can click on any component of a playbook to see more detailed information about each node.  

Reviewer: ```suggestion
1. To explore an existing playbook, click on it to open the playbook diagram in the right sidebar. Here, you can view the individual nodes and sequences, providing insight into the types of actions and structures you can create with custom playbooks. Playbooks can include multiple actions and branching conditions to handle different action sequences based on specific criteria. Click any playbook component to see detailed information about each node.
```

1. Before we start adding actions to our playbook, we’ll want to set up the initial configuration of the playbook so we get the proper inputs from the Cloud SIEM Insight.  
   1. Mouse over the Start node, and click on the Edit (pencil) icon.  
   1. In the Edit Node popup, select **Insight** from the playbook input parameters dropdown. Choosing **Insight** will automatically populate the popup view with a number of input parameters that will be added to the playbook from the corresponding Insight.
   1. Click **Update** to save and close the input parameters.
1. Add an action node:
   1. Click the **+** button on the Start node.
   1. Select **Action** from the node type options. 
   1. Fill in a node name, for example, “Get Insight Details”.
   1. From the **Integration** options, select **Sumo Logic Cloud SIEM**.
   1. For **Type**, ensure **Enrichment** is selected.
   1. As the **Action**, select **Get Insight V2**.
   1. As the **Insight ID**, select **Insight ID**.
   1. Click **Create**.
1. Add another action to the playbook by clicking the **+** icon on the **Get Insight Details** node you just created and selecting **Action**. Use the parameters outlined below:

>     1. **Name**: “Get VirusTotal Info”
>     1. **Integration**: **VirusTotal**
>     1. **Type**: **Enrichment**
>     1. **Action**: **IP Reputation**

Reviewer: ```suggestion
    1. **Name**: “Get VirusTotal Info”.
    1. **Integration**: **VirusTotal**.
    1. **Type**: **Enrichment**.
    1. **Action**: **IP Reputation**.
```

    1. For the **IP** field, click the cog icon on the right, and select the **Get Insight Details** action. Then find the **output.entity.ip.address** field and select it.
    1. Click **Create** to save the new action.
1. Add another action to the playbook by clicking the **+** icon on the **Get VirusTotal Info** node you just created and selecting **Action**. Use the parameters outlined below:

>     1. **Name**: “Add Entity Enrichment”
>     1. **Integration**: **Sumo Logic Cloud SIEM Internal**
>     1. **Type**: **Notification**
>     1. **Action**: **Add Entity Enrichment**
>     1. **Entity ID**: Click the cog icon on the right, and select the **Get Insight Details** action. Then find the **output.entity.id** field and select it.
>     1. **Enrichment Name**: “VirusTotal IP Reputation”

Reviewer: ```suggestion
    1. **Name**: "Add Entity Enrichment".
    1. **Integration**: **Sumo Logic Cloud SIEM Internal**.
    1. **Type**: **Notification**.
    1. **Action**: **Add Entity Enrichment**.
    1. **Entity ID**: Click the cog icon on the right, and select the **Get Insight Details** action. Then find the **output.entity.id** field and select it.
    1. **Enrichment Name**: “VirusTotal IP Reputation”.
```

    1. **Raw JSON**: Click the cog icon, select **Get VirusTotal Info**, then select **output.raw**.
    1. You can leave the other fields blank. Click **Create** to save the action.
1. Playbooks also allow condition nodes that can switch execution branches depending on the true/false results of a given expression. Let’s add a condition node to our playbook that will differentiate the execution branch depending on the severity of the insight.
   1. Click the **+** icon under our last action (the blue **Add Entity Enrichment** action). Choose the **Condition** node type.
   1. For the top **select a value**, select **Get Insight Details**, and then **output.severity** option. 
   Make sure **==** is selected in the middle row.
   1. For the bottom **select a value** field, add a manual value: **High**.
   1. Click **Create** to save the Condition node.
1. Click the **+** icon under the **Condition** node to create a new node. Select **Action** for this new node. Use the parameters outlined below:

>    1. **Name**: "Send Notification Email"
>    1. **Integration**: **Basic Tools**
>    1. **Type**: **Notification**

Reviewer: ```suggestion
   1. **Name**: "Send Notification Email".
   1. **Integration**: **Basic Tools**.
   1. **Type**: **Notification**.
```

In this section, we’ll create a custom incident template. This template will automatically assign the playbook you created earlier to certain new incidents, and then automatically run it.  

1. [**Classic UI**](/docs/cloud-soar/overview#classic-ui). Click the gear icon <img src={useBaseUrl('img/cloud-soar/cloud-soar-settings-icon.png')} alt="Settings menu icon" style={{border: '1px solid gray'}} width="25"/> in the top right, select **Automation**, and then select **Incident templates** in the left nav bar. <br/>[**New UI**](/docs/cloud-soar/overview#new-ui). In the main Sumo Logic menu select **Automation > Template**. You can also click the **Go To...** menu at the top of the screen and select **Template**.
1. Near the top, click the **+** icon to create a new template.

> 1. In the **Name** provide a name for the template. 

Reviewer: ```suggestion
1. In the **Name** field, provide a name for the template. 
```

> 1. In the **Category** field enter a category, for example, **Test**. 

Reviewer: ```suggestion
1. In the **Category** field, enter a category (for example, **Test**). 
```

1. Click the **+** icon next to **Playbook** to add a new playbook.
1. Select the check mark next to the playbook you created in the previous section. 
1. Click **Add**.
1. Toggle the **Autorun** switch to the **Enabled** (blue) position.

#### Create a custom automation rule

Automation rules define what happens when data is received from a source. These rules allow specific data to be parsed from the incoming data sources and then acted upon automatically. For more information, see [Automation rules](/docs/cloud-soar/automation/#automation-rules).

Automation rules can automatically pull information from sources. They can also execute playbooks based on certain criteria. For example, you might want to pull Insights from Cloud SIEM once an hour and create incidents from them. Then, you could configure a rule that runs a playbook based on the tags in the incident. This way, the entire incident response cycle is automated: Cloud SIEM identifies a threat, and playbooks in Cloud SOAR are automatically deployed to contain and eradicate the threats, restore systems, and email a final report for an analyst to review.  

> Let's create a custom automation rule. This rule will pull information from Cloud SIEM every five hours.

Reviewer: Although I don’t think we have an official style guide entry on this, I believe our convention is to write all numbers as numerals rather than spelling them out. (As an AP Stylebook follower, it pains me to say that!).

```suggestion
Let's create a custom automation rule. This rule will pull information from Cloud SIEM every 5 hours.
```
